test/Makefile.inc

```diff
@@ -17,6 +17,9 @@ INCFLAGS = \
 
 AM_CPPFLAGS = $(INCFLAGS)
 AM_CFLAGS = $(CUNIT_CFLAGS)
+if cunit_out_xml
+AM_CFLAGS  += -DTEST_CUNIT_XML
+endif
```

Comment:
This looks like an overkill. I'd suggest to use Autoconf variable rather than Automake conditional here.